fiptool fails at runtime due to missing libcrypto.so.3 during trusted-firmware-a-renesas build

While building trusted-firmware-a-renesas_2.9.0.bb in the Yocto-based rz-community-bsp, the fiptool binary fails to execute during the build process due to a missing runtime dependency:

```
| Built fiptool successfully
|
| tools/fiptool/fiptool create --soc-fw /mnt/big_dr/yocto/torizon2/rz-community-platform/build/tmp/work/smarc_rzv2l-tzn-linux/trusted-firmware-a-renesas/2.9.0/build/v2l/release/bl31.bin --nt-fw /mnt/big_dr/yocto/torizon2/rz-community-platform/build/tmp/deploy/images/smarc-rzv2l/u-boot.bin /mnt/big_dr/yocto/torizon2/rz-community-platform/build/tmp/work/smarc_rzv2l-tzn-linux/trusted-firmware-a-renesas/2.9.0/build/v2l/release/fip.bin
| tools/fiptool/fiptool: error while loading shared libraries: libcrypto.so.3: cannot open shared object file: No such file or directory
| make: *** [Makefile:1617: /mnt/big_dr/yocto/torizon2/rz-community-platform/build/tmp/work/smarc_rzv2l-tzn-linux/trusted-firmware-a-renesas/2.9.0/build/v2l/release/fip.bin] Error 127
```
Build Environment:

BSP: rz-community-bsp

Recipe: trusted-firmware-a-renesas_2.9.0.bb

Machine: smarc_rzv2l

TFA Version: v2.9.0

Host machine: Ubuntu 20.04

Possible Cause:

The fiptool binary is built and linked successfully but fails to execute because the required libcrypto.so.3 shared library is not found in the runtime library path during the build. This is likely due to it not being picked up from the native sysroot.

Proposed Fix / Workaround:

To work around this, I added a do_compile:prepend() step to explicitly export the correct LD_LIBRARY_PATH from the native sysroot before fiptool is called in rz-community-bsp/meta-rz-bsp/recipes-bsp/trusted-firmware-a/trusted-firmware-a-renesas_2.9.0.bb:
```

do_compile:prepend() { 
    export LD_LIBRARY_PATH="${WORKDIR}/recipe-sysroot-native/usr/lib:${LD_LIBRARY_PATH}" 
}
```
This ensures the shared libraries are available at runtime during the build step that executes fiptool.
I have tried this fix and build successfully

Request:

Could you confirm if this workaround is appropriate or if there's a more canonical way to ensure native build tools (like fiptool) can locate their shared libraries from the Yocto-native sysroot?

Thanks in advance!

--- a/meta-rz-bsp/recipes-bsp/trusted-firmware-a/trusted-firmware-a-renesas_2.9.0.bb
+++ b/meta-rz-bsp/recipes-bsp/trusted-firmware-a/trusted-firmware-a-renesas_2.9.0.bb
@@ -6,3 +6,14 @@ COMPATIBLE_MACHINE = "(rzg2h-family|rzg2l-family)"
 SRCREV_tfa = "cc18695622e5637ec70ee3ae8eb5e83b09d13804"
 LIC_FILES_CHKSUM += "file://docs/license.rst;md5=b2c740efedc159745b9b31f88ff03dde"
 SRC_URI = "git://github.com/renesas-rz/rzg_trusted-firmware-a.git;branch=v2.9/rz;protocol=https;name=tfa"
+
+# In TF-A v2.10.0, the variable containing linker arguments in the fiptool
+# Makefile was changed from `LDLIBS` to `LDOPTS`.
+#
+# When building with the Yocto scarthgap branch, the trusted-firmware-a.inc
+# file in meta-arm handles modification of the LDOPTS variable for TF-A 2.10.0
+# or later. So to fix the build with Scarthgap and TF-A 2.9.0 we need to modify
+# the LDLIBS variable ourselves.
+do_compile:prepend() {
+	sed -i '/^LDLIBS/ s,$, \$\{BUILD_LDFLAGS},' ${S}/tools/fiptool/Makefile
+}

--- a/meta-rz-bsp/recipes-bsp/trusted-firmware-a/trusted-firmware-a_2.10.8.bb
+++ b/meta-rz-bsp/recipes-bsp/trusted-firmware-a/trusted-firmware-a_2.10.8.bb
@@ -13,3 +13,14 @@ SRC_URI = "git://git.trustedfirmware.org/TF-A/trusted-firmware-a.git;protocol=ht
 SRC_URI:remove = "file://ssl.patch"
 
 LIC_FILES_CHKSUM += "file://docs/license.rst;md5=b2c740efedc159745b9b31f88ff03dde"
+
+# In TF-A v2.10.0, the variable containing linker arguments in the fiptool
+# Makefile was changed from `LDLIBS` to `LDOPTS`.
+#
+# When building with the Yocto kirkstone branch, the trusted-firmware-a.inc
+# file in meta-arm handles modification of the LDLIBS variable for TF-A 2.9.0
+# or earlier. So to fix the build with Kirkstone and TF-A 2.10.8 we need to
+# modify the LDOPTS variable ourselves.
+do_compile:prepend() {
+	sed -i '/^LDOPTS/ s,$, \$\{BUILD_LDFLAGS},' ${S}/tools/fiptool/Makefile
+}

